exec/java-exec/src/main/codegen/templates/CastFunctions.java

```diff
@@ -54,14 +56,14 @@ public void setup() {}
   public void eval() {
     <#if (type.from.startsWith("Float") && type.to.endsWith("Int"))>
     boolean sign = (in.value < 0);
-    in.value = java.lang.Math.abs(in.value);
-    ${type.native} fractional = in.value % 1;
+    ${type.native} value = java.lang.Math.abs(in.value);
+    ${type.native} fractional = value % 1;
```

Comment:
Thanks for the pointing this. Drill uses `half away from zero` rounding mode, so method `Math.round()` wouldn't help. I rewrote current code and now it handles boundary cases in the same way as the `Math.round()`.